test/mpdecimal_test.exs

```diff
@@ -2,27 +2,79 @@ defmodule MPDecimalTest do
   use ExUnit.Case
   doctest MPDecimal
 
-  test "correct digit handling" do
-    result = MPDecimal.Nif.power("3.14159", "1.58732")
-    assert result == "6.153686306332765322891030633"
+  test "positive integer exponents" do
+    assert_exp(2, 2, "4.0")
+    assert_exp(2, 3, "8.0")
+    assert_exp(1, 8, "1.0")
+    assert_exp(2, 8, "256.0")
+    assert_exp(2, 0, "1.0")
+    assert_exp(256, 0, "1.0")
   end
 
-  test "Strange NaN behavior???" do
-    result = MPDecimal.Nif.power("3.1", "1.587321")
-    assert result == "6.1536933506402224364507308462"
+  test "negative integer exponents" do
+    assert_exp(-1, 2, "1.0")
+    assert_exp(-2, 3, "-8.0")
+    assert_exp(-1, 0, "1.0")
+    assert_exp(-1, -1, "-1.0")
+    assert_exp(-2, -2, "0.25")
   end
 
-  test "bad args do not crash the VM" do
-    assert_raise ArgumentError, fn ->
-      MPDecimal.Nif.power("12.34", nil)
-    end
+  test "handling zero base values" do
+    assert_exp(0, 1, "0.0")
+    assert_exp(0, 2, "0.0")
 
-    assert_raise ArgumentError, fn ->
-      MPDecimal.Nif.power(false, "12.34")
-    end
+    assert_exp("0", 1, "0.0")
+    assert_exp("0", 2, "0.0")
+  end
+
+  test "decimal base values with integer powers" do
+    assert_exp("0.5", 2, "0.25")
+    assert_exp("0.5", 0, "1")
+    assert_exp("0.5", -1, "2")
+    assert_exp("0.5", -2, "4")
+    assert_exp("-0.5", 2, "0.25")
+    assert_exp("-0.5", 3, "-0.125")
+    assert_exp("-0.5", 0, "1")
+    assert_exp("-0.5", -1, "-2")
+    assert_exp("-0.5", -2, "4")
+    assert_exp("-0.5", -3, "-8")
+  end
+
+  test "integer base values with decimal powers" do
+    assert_exp(2, "2.5", "5.656854249492380195206754897")
+    assert_exp(2, "1.75", "3.363585661014858172124501905")
+    assert_exp(2, "0.9", "1.866065983073614831962686532")
+    assert_exp(2, "0.5", "1.414213562373095048801688724")
+    assert_exp(2, "0", "1")
+    assert_exp(2, "-0.5", "0.7071067811865475244008443621")
+    assert_exp(2, "-1.5", "0.3535533905932737622004221811")
+    assert_exp(0, "2.5", "0")
+    assert_exp(0, "1.75", "0")
+    assert_exp(0, "0.9", "0")
+    assert_exp(0, "0.5", "0")
+  end
+
+  test "infinity cases" do
+    assert_exp(0, -1, "Infinity")
+    assert_exp("0.0", "-0.5", "Infinity")
+    assert_exp("Infinity", 0, 1)
+  end
+
+  test "error cases" do
+    assert_exp_error("0", "0", "[IEEE_Invalid_operation]")
+    assert_exp_error("-2", "2.5", "[IEEE_Invalid_operation]")
```

Comment:
raising a negative base to a decimal power works in Wolfram Alpha, but fails here. I think that should be fine for our use-case, but I was surprised by the discrepancy. Not sure if we care about that.

Reply:
Note that since the context is configured to behave as close as possible to the Decimal library, that it just spits out IEEE_Invalid_Operation for nearly every single error. There is a way to get finer-grained signals if you think that would be helpful.

Reply:
I think that the behavior here makes sense. If the base is negative and the exponent is irrational, then the result is not a real number.

Reply:
👍 that makes sense to me why this is failing for negative bases and negative powers. I think following the lead of the mpdecimal library on this makes total sense.

Getting more specific errors could be nice in some ways, but it definitely seems like something we would want to improve in a 1.0 release, not super high priority right now.